src/test/java/io/numaproj/numaflow/function/reduce/ReduceSupervisorActorTest.java

```diff
@@ -0,0 +1,148 @@
+package io.numaproj.numaflow.function.reduce;
+
+import akka.actor.ActorRef;
+import akka.actor.ActorSystem;
+import akka.pattern.Patterns;
+import com.google.protobuf.ByteString;
+import io.numaproj.numaflow.function.Datum;
+import io.numaproj.numaflow.function.Function;
+import io.numaproj.numaflow.function.Message;
+import io.numaproj.numaflow.function.ReduceOutputStreamObserver;
+import io.numaproj.numaflow.function.metadata.IntervalWindowImpl;
+import io.numaproj.numaflow.function.metadata.Metadata;
+import io.numaproj.numaflow.function.metadata.MetadataImpl;
+import io.numaproj.numaflow.function.v1.Udfunction;
+import org.junit.Test;
+import scala.concurrent.Await;
+import scala.concurrent.Future;
+import scala.concurrent.duration.Duration;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeoutException;
+
+import static org.junit.Assert.assertEquals;
+
+public class ReduceSupervisorActorTest {
```

Comment:
This is great as it ensures a high coverage already. I searched around and see a lot of articles/blogs suggesting using Akka Testkit to asynchronous test actors in a controlled but realistic environment.

Have you considered using it? https://doc.akka.io/docs/akka/current/testing.html

Reply:
The tests I have written should cover most of the cases. I didn't find any need for using Akka's testing framework. But honestly, I didn't spend much time on unit tests. I think we can definitely refactor the tests in upcoming prs using Akka's testing framework.

Reply:
please open an issue to review Akka's testing strategy.

Reply:
#20